frontend/src/App.js

```diff
@@ -1,17 +1,25 @@
 import { Route, Switch } from 'react-router'
 import CourseDetails from './pages/CourseDetails'
 import CourseOverview from './pages/CourseOverview'
+import Login from './pages/Login'
+import ProtectedRoute from './auth/ProtectedRoute'
+import AuthProvider from './auth/AuthProvider'
 
 function App() {
   return (
-    <Switch>
-      <Route exact path="/">
-        <CourseOverview />
-      </Route>
-      <Route exact path="/course/:courseName">
-        <CourseDetails />
-      </Route>
-    </Switch>
+    <AuthProvider>
+      <Switch>
+        <Route exact path="/login">
+          <Login />
+        </Route>
+        <ProtectedRoute exact path="/home">
+          <CourseOverview />
+        </ProtectedRoute>
+        <ProtectedRoute exact path="/course/:courseName">
+          <CourseDetails />
+        </ProtectedRoute>
```

Comment:
`exact` is not needed here. 

Btw, what do you think about 
```javascript
<Route path="/home" component={Home} />
```
short and sweet :)